test/functional/cmdLineTests/valuetypeddrtests/unflattenedvaluetypeddrtests.xml

```diff
@@ -0,0 +1,132 @@
+		<saveoutput regex="no" type="required" saveName="valueAddr2" splitIndex="1" splitBy="= !j9object " showMatch="yes">[1] = !fj9object 0x</saveoutput>
+		<saveoutput regex="no" type="required" saveName="arrayAddr" splitIndex="1" splitBy="= !j9object " showMatch="yes">[2] = !fj9object 0x</saveoutput>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $valueAddr1$ to show the contents of the reference object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $valueAddr1$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">QValueInt; i = !fj9object 0x</output>
+		<saveoutput regex="no" type="required" saveName="intAddr1" splitIndex="4" splitBy=" " showMatch="yes">QValueInt; i = !fj9object 0x</saveoutput>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !fj9object $intAddr1$ to show the int contents of the reference object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!fj9object $intAddr1$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0x7FFFFFFF (offset = 0) (ValueInt)</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $valueAddr2$ to show the contents of the reference object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $valueAddr2$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">QValueInt; i = !fj9object 0x</output>
+		<saveoutput regex="no" type="required" saveName="intAddr2" splitIndex="4" splitBy=" " showMatch="yes">QValueInt; i = !fj9object 0x</saveoutput>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !fj9object $intAddr2$ to show the int contents of the reference object">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!fj9object $intAddr2$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">I i = 0xB670C61E (offset = 0) (ValueInt)</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+
+	<test id="Run !j9object $arrayAddr$ to display array contents">
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core $DUMPFILE$</arg>
+			<input>!j9object $arrayAddr$</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success" showMatch="yes">!j9object $valueAddr1$</output>
+		<output regex="no" type="success" showMatch="yes">!j9object $valueAddr2$</output>
+		<output regex="no" type="failure">Problem running command</output>
+	</test>
+	<!-- TODO: add command line test for flatobject once it is functional for unflattened objects -->
```

Comment:
add the `!flatobject` tests here as well, it should have reasonable behaviour even if the type is not flattened

Reply:
currently, they're not working as intended, rather they are just doing the same thing as `!j9object <addr>` and the output has extra unintended spacing. There needs to be work done to recursively call all the commands to show all the fields, similar to what happens for reference types with flattening enabled

Reply:
Thats fine, in fact, for non value types it should always only show one level otherwise there is a risk of cycles